spec/requests/schoolings_controller_spec.rb

```diff
@@ -18,4 +34,34 @@
              params: { confirmed_director: "1" }
     end
   end
+
+  describe "retry_eligibile_payment_requests" do
+    before do
+      schooling.update!(end_date: Date.parse("2024-06-21"))
+    end
+
+    context "when the payment request is retry eligible" do
+      it "Do not return abrogated decision error" do
+        delete abrogate_decision_school_year_class_schooling_path(schooling.classe.school_year,
+                                                                  class_id: schooling.classe.id, id: schooling.id),
+               params: { confirmed_director: "1" }
+        expect(payment_request.last_transition.metadata).not_to include(error_message)
+      end
+    end
+
+    context "when the payment request is not retry eligible" do
+      let(:pfmp) { create(:pfmp, :completed) }
+
+      before do
+        schooling.update!(pfmps: [pfmp])
+      end
+
+      it "Return abrogated decision error" do
+        delete abrogate_decision_school_year_class_schooling_path(schooling.classe.school_year,
+                                                                  class_id: schooling.classe.id, id: schooling.id),
+               params: { confirmed_director: "1" }
+        expect(payment_request.current_state).not_to eq(:ready)
```

Comment:
Ici il faudrait plutôt avoir un test similaire à celui du dessus. Parce que au final ce test n'a pas vraiment de valeur pour nous tel quel.

`expect(payment_request.last_transition.metadata).to include(error_message)`

Reply:
Si je veux tester comme ca il faut que je créé soit un cas qui ne passe pas dans cette condition :
@schooling.pfmps.in_state(:validated)
Soit dans le cas ou la payment request n'est pas incomplete donc sans message d'erreur
Soit que la payment request ne contienne pas l'erreur au départ.

J'ai préféré vérifier la première condition, les autres cas ne me paraissaient pas forcément valables

Reply:
Je comprends pas ce que tu veux dire. Je parle d'un test qui partirait d'une p_r incomplete pour une raison autre que l'abrog de da et dans le test on vérifie bien que le message reste le même (tu peux utiliser la fonction `change` de RSpec)

Reply:
A regarder dans : #1049